src/time/Storage.cpp

```diff
@@ -26,9 +26,9 @@ void Storage::setSampleAtTime(double time, Sample sample)
   PRECICE_ASSERT(math::smallerEquals(WINDOW_START, time), "Setting values outside of valid range!");
   PRECICE_ASSERT(math::smallerEquals(time, WINDOW_END), "Sampling outside of valid range!");
   // check if key "time" exists.
-  auto existingStample = std::find_if(_stampleStorage.begin(), _stampleStorage.end(), [&time](const auto &s) { return math::equals(s.timestamp, time); });
-  if (existingStample == _stampleStorage.end()) { // key does not exist yet
-    PRECICE_ASSERT(math::smaller(maxStoredNormalizedDt(), time), maxStoredNormalizedDt(), time, "Trying to write values with a time that is too small. Please use trim() or clear(), if you want to write new samples to the storage.");
+  auto existingSample = std::find_if(_stampleStorage.begin(), _stampleStorage.end(), [&time](const auto &s) { return math::equals(s.timestamp, time); });
+  if (existingSample == _stampleStorage.end()) { // key does not exist yet
+    PRECICE_ASSERT(math::smaller(maxStoredNormalizedDt(), time), maxStoredNormalizedDt(), time, "Trying to write values with a time that is too small. Please use clear(), if you want to write new samples to the storage.");
```

Comment:
Is `math::smaller` here the right thing to do? Shouldn't it be `not math::larger`?

Reply:
No. Technically `not math::greater` and `math::smallerEquals` are the same, but if we have `time == mexStoredNormalizedDt()` we would overwrite and not append. So the other branch of the `if`.